packages/di/src/lib/provider.ts

```diff
@@ -1,12 +1,20 @@
 import { Injector } from './injector.js';
 
 export class StaticToken<T> {
-  name;
-  factory;
+  #name;
+  #factory;
+
+  get name() {
+    return this.#name;
+  }
+
+  get factory() {
+    return this.#factory;
+  }
 
   constructor(name: string, factory?: () => T) {
```

Comment:
minor: could also use `public readonly` here instead of declaring getters

Reply:
that only makes it readonly at build time and I want this to be readonly at runtime

Reply:
Oh I see, that would actually be stripped... and one can also use the library without the typings as vanillajs...